en/companion_computer/pixhawk_rpi.md

```diff
@@ -0,0 +1,259 @@
+# Using the Raspberry Pi with the Pixhawk Controllers
+
+This section describes how to connect the Pixhawk to the Raspberry Pi ("RPi"). Both the software setup and the wire connections between the hardware are presented. 
+
+There are different ways to connect the RPi with the Pixhawk:
+
+1. Via Ethernet port.
+
+2. Serial connection: via `TELEM` port or via `USB` port.
+
+The connection via `TELEM` port can be done with a FTDI Chip USB-to-serial adapter board or directly at the RPi's TX/RX pins. The following steps describes the connections using the TX/RX pins, since no additional board is required.
+
+There are different models of Pixhawk and RPi, therefore, the main concepts are covered and illustrated in order to give the user the necessary knowledge for any other configuration. The first step is the identification of the `TELEM 2` and the `USB` ports in the Pixhawk. The image illustrates the ports in the Pixhawk-6C Model.
+
+<img src="../../assets/companion_computer/pixhawk_rpi/pixhawk_6c.png" width="402" height="452" />
+
+## PX4 setup on Pixhawk
+
+Connect the Pixhawk with the laptop/desktop via `USB` port and check, at the QGroundControl, if the firmware (PX4) is updated. 
+The firmware update can be done in the QGroundControl or following the next steps:
+
+Run in the terminal:
+
+```
+sudo apt -y install gcc-arm-none-eabi make stlink-tools
+```
+
+Clone the last PX4 version and run the script setup:
+
+```
+git clone https://github.com/PX4/PX4-Autopilot.git --recursive 
+PX4-Autopilot/Tools/setup/ubuntu.sh
+```
```

Comment:
This is all basically building PX4 on Ubuntu Linux right? 
1. PX4 should be build using a specific build toolchain, as per https://github.com/PX4/PX4-Autopilot/blob/main/Tools/setup/ubuntu.sh#L162 - that's basically a link to http://docs.px4.io/main/en/dev_setup/dev_env_linux_ubuntu.html#simulation-and-nuttx-pixhawk-targets
2. What are stlink-tools for?

Reply:
Or to put it another way, there should be no need to build PX unless you want to change the messages supported. Generally you could select the "main/master" firmware as in QGC right? http://docs.px4.io/main/en/config/firmware.html#installing-px4-master-beta-or-custom-firmware

But I'm still interested in why you would want stlink-tools :-)

Reply:
> This is all basically building PX4 on Ubuntu Linux right?

 yes 

> 1. PX4 should be build using a specific build toolchain, as per https://github.com/PX4/PX4-Autopilot/blob/main/Tools/setup/ubuntu.sh#L162 - that's basically a link to http://docs.px4.io/main/en/dev_setup/dev_env_linux_ubuntu.html#simulation-and-nuttx-pixhawk-targets

I prefer to write here most of the steps, when possible, instead of links, so the user can follow all in the sequence. If you prefer I can change it for the link, but if possible, I can complement. 
What did I miss? Should it be:

PX4-Autopilot/Tools/setup/ubuntu.sh --no-sim-tools (since nuttx by default is installed?)

> 2. What are stlink-tools for?

I look I long time for it when doing the build. I am not sure why it is necessary, but I found in the PX4 forum that it is required, so after install it works.

Reply:
> No, this is building the PX4 in the Pixhawk.

it is not possible to build PX4 "in" the Pixhawk. So I think you mean "for" the Pixhawk.

What we will do here is link to the normal PX4 instructions, then add a summary of the steps `PX4-Autopilot/Tools/setup/ubuntu.sh --no-sim-tools`.

Last of all perhaps we will add a note about  stlink-tools as well.

In general you should not need these tools. The main reason would be to do on target debugging. Though in some cases you can use them to manually flash bootloaders and stuff. I don't think most people would need them - what did you see when you did not add these?

Reply:
Please let me know if my answer below (https://github.com/PX4/PX4-user_guide/pull/2318#discussion_r1123837322) have clarified it.